tools/baseimage/pkg/gce/helper.go

```diff
@@ -139,6 +143,38 @@ func (h *GceHelper) CreateImage(ins, disk, name string) error {
 	return h.waitForGlobalOperation(op)
 }
 
+func (h *GceHelper) WaitForInstance(name string) error {
+	for attempt := 0; attempt < 3; attempt++ {
+		time.Sleep(30 * time.Second)
+		log.Printf("wait for instance: uptime attempt number: %d", attempt)
+		if err := h.RunCmdOnInstance(name, "uptime"); err == nil {
+			return nil
+		}
+	}
+	return errors.New("waiting for instance timed out")
+}
+
+func (h *GceHelper) UploadBashScriptOnInstance(insName, scriptName, scriptContent string) error {
```

Comment:
The GCE Helper API is a wrapper around the GCE API performing CRUD operations. The `waitForInstance`, `uploadBashScript` and `runCmd` don't follow such pattern and that's the reason they are not part of the GCE Helper API in first place.

You can move them to the helper.go as public functions, but don't make them part of the GCE Helper class at least in this PR.